docs/interaction_based_testing.adoc

```diff
@@ -792,10 +792,20 @@ SubscriberImpl subscriber = Spy(constructorArgs: ["Fred"])
 
 A spy is always based on a real object. Hence you must provide a class type rather
 than an interface type, along with any constructor arguments for the type.
-If no constructor arguments are provided, the type's default constructor will be used.
+If no constructor arguments are provided, the type's no-arg constructor will be used.
+
+If the given constructor arguments lead to an ambiguity, you can cast the constructor
+arguments as usual using `as`. If the testee for example has one constructor with a
```

Comment:
both `x as String` and `(String) x` should work with the new code right? Both variants should be mentioned

Reply:
Yes, those are treated the same by the parser. I changed the tests to test both variants and the docs to mention both.